sdk/core/Azure.Core/src/RequestContext.cs

```diff
@@ -52,5 +64,90 @@ public void AddPolicy(HttpPipelinePolicy policy, HttpPipelinePosition position)
             Policies ??= new();
             Policies.Add((position, policy));
         }
+
+        /// <summary>
+        /// Adds a custom classifier to the <see cref="ResponseClassifier"/> that decides if the response
+        /// received from the service should be considered an error response, for this service call.
+        /// The custom classifier is applied before the default classifier.
+        /// This is useful in cases where you'd like to prevent specific response status codes from appearing as errors in
+        /// logging and distributed tracing.  It will also prevent the call from throwing an exception when a response with
+        /// this status code is received.
+        /// </summary>
+        /// <param name="statusCode">The status code to customize classification for.</param>
+        /// <param name="isError">Whether the passed-in status code should be classified as an error.</param>
+        public void AddClassifier(int statusCode, bool isError)
+        {
+            int length = _classifiers == null ? 0 : _classifiers.Length;
```

Comment:
Could we just have at most one classifier in the context? i.e. no array. The way it would work is the following: 
1. by default the classifier field is null
2. If the user calls AddClassfier(int, bool) method we assign StatusClassifier to the field. 
3. If the user calls AddClassifier(MessageClassfier) we switch to CompositeClassifier.

Reply:
Yes, we can do that, it will simplify things a lot.  From prior discussions, though, it sounded like we wanted this "last addition has precedence" -- or did I misunderstand that?

Would we want to change "AddClassifier" to "SetClassifier"?  Or are we thinking the verb "Add" means "add it to the composed classifier", not that Set would mean "set it on the requestcontext?

Reply:
This does not preclude the "last addition has precedence" semantics, does it?

One thing that does preclude them is the optimization to store all int,bool tuples in one classifier. But maybe we are ok with that, i.e. the semantics would be: classifiers added using AddClassifier(MessageClassfier) run first in the order they were called, and then all added using AddClassfier(int, bool).

Reply:
> This does not preclude the "last addition has precedence" semantics, does it?

It does if we replace the one classifier on each call to AddClassifier.  We get "last writer wins", but are unable to compose classifiers from the multiple calls.

> classifiers added using AddClassifier(MessageClassfier) run first in the order they were called, and then all added using AddClassfier(int, bool).

This sounds like relatively complex logic for library callers to understand, so I wonder about the trade-off here.

Could we back up and clarify what motivates the desire to shift to a single classifier rather than an array?  Is it a performance optimization, or to achieve the functionality differently?  Are we worried about getting a lot of StatusClassifiers in the chain?